### ISSUE TITLE
Having trouble joining my dedicated server

### ISSUE THREAD

**Issue by bov1ne** [DFJacob/AbioticFactorDedicatedServer#2]
[posted this on forums too, maybe this is a better spot]

It's hosted on another pc on the lan. I can see it by adding the LAN IP in the Server Browser within steam/see the latency and refresh it, but when I go to join it, it just launches the game. If it's in the in game server browser..I can't see it/find it (search would be nice here). I tried 'switch to lan' but it doesn't see the server there.

Looking at the log of the server, it says it's bound and listening to port 7777/doesn't look like any issues. I have that all port forwarded. firewalls disabled on both machines.

since then: 

I remote logged into my friends pc (not on my LAN) and was able to add the server using my WAN IP to their steam browser and it's the same situation. She can see the latency/refresh the [server..name](steam://openurl_external/https://steamcommunity.com/linkfilter/?u=http%3A%2F%2Fserver..name) matches, but when you hit 'Join', all it does is launch the game and nothing. Also doesn't appear to be in the server list.

Also installed the server on the same PC I'm playing the game on/changed port forwarding. Same result though..can add server to steam browser but not join it/ see it in game otherwise.

**Comment by DFJacob** [comment]
Can you see the server in the steam client server browser by filtering Abiotic Factor in the Internet or LAN tab? Sometimes adding the IP to the favorites gets it to show but may be missing some of the steam registration requirements.

If the server is running on another machine on your LAN, are you able to view/join the server if you run it on the same machine. You can achieve this easily by downloading the server in steam and running it (Will just have a random steamid as the server name).

**Comment by bov1ne** [comment]
Steam client server browser = View -> Game Servers in the steam UI? if so, yes..it's always there in each scenario. I can see the game name, # of players (I set it for 3) and the latency. But when I use the 'join' button to go into the server, it launches the game and nothing happens from there, I just go into the main menu. I did install it on the same machine, with the same result.

**Comment by DFJacob** [comment]
Unreal also has a launch parameter for assigning the server a specific IP address called `-multihome` eg. `-multihome=192.168.1.10`. Wold that help with your VM/Ethernet issue?

Currently joining through the steam client's server browser is unsupported, we're planning on adding search/filter options for the in-game browser

**Comment by 3x00** [comment]
I have the same problem, I can find my server on steam client's server browser, but can't find it in the game browser.
by the way, I want to know the format of startup argument "-ServerPassword=" is how to replace, the password need "" or not?

**Comment by titanbubu** [comment]
@3x00 @bov1ne 
Are you guys confirmed port forwarded on both 7777 and 27015? The second port is the steam query port for showing up on game browsers.

**Comment by Oxsynn** [comment]
I am also having this issue. Notably, the server showed up last night, I didn't make any changes so I'm not sure why it showed up. But after applying the hotfix to both client and server, the server doesn't show up in the in-game, only in the steam browser if I manually add the IP. 

Some context: Server is on my LAN, doesn't show up in either tab in the ingame browser. I've forwarded the ports and confirmed that the ports are open via nmap (notably, they come up as filtered, I assume this is because I forwarded the ports as TCP/UDP, and usually UDP shows as filtered) 

I hope direct connect is a planned feature, I feel like that tends to resolve most issues like this (and it allows custom ports for added security!)

**Comment by bov1ne** [comment]
I'm not having any game stopping problems anymore now that I understand the steam browser is not currently supported and I keep that adapter disabled (I haven't tried the multihome parm yet). When I tested things now, on both my local machine and another machine on my Lan, I was able to join either server from my friend's remote location in game, and from my own PC. One thing I noticed is the server doesn't show up for me in the 'Internet' tab of the game, only in the 'LAN'. My friend was able to see/search me in Internet just fine though, but I could see someone getting confused when they see an empty list and can't join their server until they look for it in LAN.

**Comment by 3x00** [comment]
I guess the problem is my server have no public ip, I used the other vps to reverse proxy my server port 7777 and 27015, the info to steam (I gusee) is via my server's ip, and steam check to my server is impossiable, but if check the server via my vps's ip is ok.

**Comment by titanbubu** [comment]
https://www.yougetsignal.com/tools/open-ports/

If you're hosting on a Windows machine you always have to quadruple check some program is not blocking your port requests to that machine. Built in windows firewall, Anti virus, and more can get in the way of your machine getting the public IP for that port.

**Comment by 3x00** [comment]
@titanbubu no, the problem is my server doesn't have public ip, my server is in a huge LAN.

**Comment by titanbubu** [comment]
You do not need your host to be connected directly to the modem to host ie getting the entire public IP. 
Mine is in a LAN and it works just fine. 
Port forwarding essentially mimics having the public IP because that traffic is forwarded to that machine.

**Comment by titanbubu** [comment]
If you're in a managed network such as an apartment complex you have multiple routers between you and the wan IP. 
They all need to have a firewall/networking rule to forward the two ports towards your router. 
You're kind of screwed but you can call your network support line and see if they will do it. 
It's unlikely. Sorry dude. You have to get your own Internet connection.

**Comment by 3x00** [comment]
yep, so I use a vps to proxy my server's port, it works on the other game enshrouded, but that game can use steam client's server browser to connect the game server directly, so I think this is the solution.

**Comment by wkndr** [comment]
Successfully started dedicated server by systemd unit, it is visible but connection ends up by "fleming" error. Then I tried to start sh-file directly using utility named screen and successfully connected to my server. Currently can't figure out true cause

**Comment by xxlio109** [comment]
I have the same problem, I can't join a dedicated server in my LAN, but my friends can join a dedicated server in my LAN via the steam game server.

**Comment by Crazypeak** [comment]
I can't understand the design principle of the server tool. According to the server tool made by the same Unreal Engine (Conan), after setting the port monitoring, I can intuitively see the specific port forwarding on the network. However, our tool, even if it is the default port 7777, only prompts the binding success in the log, but there is no listening forwarding of port 7777 in the port list (whether it is TCP or UDP)

**Comment by Zoss0** [comment]
> yep, so I use a vps to proxy my server's port, it works on the other game enshrouded, but that game can use steam client's server browser to connect the game server directly, so I think this is the solution.

I've got the same issue. We've also played the same thing together. I use Rathole as my reverse proxy and punch my UDP traffic out to my VPS.
